pvacseq/lib/generate_fasta.py

```diff
@@ -11,70 +38,104 @@ def distance_from_start(position, string):
+    elif distance_from_start(position, full_wildtype_sequence) >= one_flanking_sequence_length and distance_from_end(position, full_wildtype_sequence) >= one_flanking_sequence_length:
+        start_position = position - one_flanking_sequence_length
+        end_position   = start_position + peptide_sequence_length
+        wildtype_subsequence = full_wildtype_sequence[start_position:end_position]
         mutation_position = one_flanking_sequence_length
     else:
         sys.exit("Something went wrong during the retrieval of the wildtype sequence at position(%s, %s, %s)" % chromsome, start, stop)
-
     return mutation_position, wildtype_subsequence
 
-
 def main(args_input = sys.argv[1:]):
-    parser = argparse.ArgumentParser("Generate Variant Sequences", description='')
-    parser.add_argument('input_file', type=argparse.FileType('r'), help='input list of variants',)
+    parser = argparse.ArgumentParser('Generate Variant Sequences', description='')
+    parser.add_argument('input_file', type=argparse.FileType('r'), help='input vcf',)
     parser.add_argument('peptide_sequence_length', type=int, help='length of the peptide sequence')
     parser.add_argument('output_file', type=argparse.FileType('w'), help='output FASTA file')
 
     args = parser.parse_args(args_input)
 
-    tsvin = csv.reader(args.input_file, delimiter='\t')
+    peptide_sequence_length = args.peptide_sequence_length
+
+    vcf_reader = vcf.Reader(args.input_file)
+    csq_format = parse_csq_format(vcf_reader)
 
+    transcript_count = {}
     pattern = re.compile('([A-Z])(\d+)([A-Z])');
-    for line in tsvin:
-        match = pattern.match(line[7]);
-        if match is not None:
-            wildtype_amino_acid, position, mutant_amino_acid = match.group(1, 2, 3)
-            position = int(position) - 1
-
-            wildtype_sequence = line[9];
-            if wildtype_amino_acid != wildtype_sequence[position]:
+    for entry in vcf_reader:
+        transcripts = parse_csq_entries(entry.INFO['CSQ'], csq_format)
+        for transcript in transcripts:
+            consequence_string = transcript['Consequence']
+            if consequence_string is None:
                 continue
-            else:
-                mutation_position, wildtype_subsequence = get_wildtype_subsequence_for_printing(position, wildtype_sequence, args.peptide_sequence_length, line)
-                mutant_subsequence = list(wildtype_subsequence)
-
-                mutant_subsequence[mutation_position] = mutant_amino_acid
-                if len(wildtype_subsequence) > 0:
-                    for designation, subsequence in zip(['WT', 'MT'], [wildtype_subsequence, ''.join(mutant_subsequence)]):
-                        fasta_header = '.'.join(['>' + designation, line[5], line[7]])
-                        args.output_file.writelines([fasta_header + "\n", subsequence + "\n"])
+            consequences = list(set(consequence.lower() for consequence in consequence_string.split('&')))
```

Comment:
I think using set comprehension (just `{ ... }` instead of `list(set( ... ))`  ) is better here for two reasons:
1)By directly initializing the set, we avoid the inefficiency of converting from generator to set to list
2)Line 110 becomes a constant time operation by searching through a set, as opposed to a linear time operation by searching a list